packaging/generic-deb/debian/postinst

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+set -e
+
+systemctl enable ecs
```

Comment:
You should actually not need this file at all. If you do include it, you should not explicitly call `systemctl` in it yourself. If you do include a postinst script, you should be sure to include `#DEBHELPER#` on a line by itself, which debhelper will fill in with the appropriate `systemctl` invocations (and anything else made necessary by its integration with the package)

Reply:
I tried to remove this file and build the package but the service is installed in a "disabled" state:
```
% sudo systemctl status ecs
● ecs.service - Amazon Elastic Container Service - container agent
     Loaded: loaded (/lib/systemd/system/ecs.service; disabled; vendor preset: enabled)
     Active: inactive (dead)
       Docs: https://aws.amazon.com/documentation/ecs/
```

Is there some way to only enable the service without a postinst script?

I tried using `dh ... --with systemd` in the rules makefile, but that also auto-starts the service, which we do not want.

Reply:
Expected behavior on Debian-derived systems is to enable and start the service upon installation. However, if you really don't want that, you can use the `--no-start` option to `dh_installsystemd`. Put the following in your `override_dh_auto_install` recipe in `debian/rules`:
```
	cp ecs.service debian/
	dh_installsystemd --name ecs --no-start
	dh_systemd_enable
```
With this in place, you should see quite a bit of content inserted into the postinst, prerm, and postrm scripts in the binary package. This will take care of all the complexity around service management across package upgrades, removals, etc.

If you're not going to start it, then I also recommend that you don't mark it as enabled, either. Use the `--no-enable` flag to `dh_installsystemd`.  Having a service installed and enabled but not actually running is rarely an expected scenario.

You should also update the `clean` recipe to remove the `debian/ecs.service` file you created there, and you can remove `ecs.service` from `amazon-ecs-init.install`

Reply:
this is done.

We are enabling the service because we want it to autostart when the instance is restarted. We don't want it to start, however, because the user needs to be able to setup their config file before starting the service. As in, the normal installation process would look like:

```
apt install -y ./amazon-ecs-init.deb
echo "ECS_CLUSTER=myCluster" > /etc/ecs/ecs.config
systemctl start ecs
```

Reply:
Instead of `systemctl start ecs`, you could recommend `systemctl enable --now ecs`. This lets you install the unit file in a stopped and disabled state, which is more consistent with the workflow you expect. Leaving it enabled means that it'll start if the instance is rebooted for some reason, even if the user hasn't finished configuring it.

Reply:
that's a good point, will make the change